java/grpc-client/src/test/java/com/youtube/vitess/client/grpc/GrpcClientTlsTest.java

```diff
@@ -0,0 +1,203 @@
+        IOUtils.copy(GrpcClientTlsTest.class.getResourceAsStream("/cert-template.config"), writer, "UTF-8");
+        final String content = writer.toString();
+        content.replaceAll("%s", commonName);
+        java.nio.file.Files.write(Paths.get(config), content.getBytes("UTF-8"));
+
+        final Process createKeyAndCSR = new ProcessBuilder(
+                // openssl req -newkey rsa:2048 -days 3600 -nodes -batch -config cert-template.config -keyout vtgate-client-key.pem -out vtgate-client-req.pem
+                Arrays.asList(
+                        "openssl",
+                        "req",
+                        "-newkey",
+                        "rsa:2048",
+                        "-days",
+                        "3600",
+                        "-nodes",
+                        "-batch",
+                        "-config",
+                        config,
+                        "-keyout",
+                        key,
+                        "-out",
+                        req)
+        ).start();
+        createKeyAndCSR.waitFor();
+
+        final Process askHarshit = new ProcessBuilder(
+                // openssl rsa -in vtgate-client-key.pem -out vtgate-client-key.pem
+                Arrays.asList(
+                        "openssl",
+                        "rsa",
+                        "-in",
+                        key,
+                        "-out",
+                        key)
+        ).start();
+        askHarshit.waitFor();
+
+        final Process signKey = new ProcessBuilder(
+                // openssl x509 -req -in vtgate-client-req.pem -days 3600 -CA ca-cert.pem -CAkey ca-key.pem -set_serial 04 -out vtgate-client-cert.pem
+                Arrays.asList(
+                        "openssl",
+                        "x509",
+                        "-req",
+                        "-in",
+                        req,
+                        "-days",
+                        "3600",
+                        "-CA",
+                        caCert,
+                        "-CAkey",
+                        caKey,
+                        "-set_serial",
+                        serial,
+                        "-out",
+                        cert)
+        ).start();
+        signKey.waitFor();
+    }
+
+    private static void startVtgate() throws Exception {
+        String vtRoot = System.getenv("VTROOT");
+        if (vtRoot == null) {
+            throw new RuntimeException("cannot find env variable VTROOT; make sure to source dev.env");
+        }
+
+        ServerSocket socket = new ServerSocket(0);
+        port = socket.getLocalPort();
+        socket.close();
+
+        vtgateclienttest =
+                new ProcessBuilder(
+                        Arrays.asList(
+                                vtRoot + "/bin/vtgateclienttest",
+
+// TODO: Uncomment the following lines once the corresponding client-side changes are added to "createClientConnection"
```

Comment:
Tied in with my comment toward @harshit-gangal and @alainjobart above.  I'm not sure if I'm really using the right flags and artifacts here.